daemon/heartbeat.c

```diff
@@ -832,6 +831,62 @@ hb(
 }
 
 
+//
+//  NAME:
+//
+//      hb_send
+//
+//  DESCRIPTION:
+//
+//      The heartbeat sending thread.
+//
+//  FORMAL PARAMETERS:
+//
+//          
+//  RETURN VALUE:
+//
+//
+//  ENVIRONMENT:
+//
+//
+
+MTC_STATIC  void *
+hb_send(
+    void *ignore)
+{
+    MTC_BOOLEAN         term = FALSE;
+    MTC_CLOCK           last, now, wait;
+
+    log_message(MTC_LOG_INFO, "HB_send: thread ID: %ld.\n", syscall(SYS_gettid));
+    do
+    {
+        // check fist
+        hb_check_fist_sticky();
+
+        last = _getms();
+
+        // send heartbeat
+        send_hb();
+
+        now = _getms();
+        wait = (_t1 * ONE_SEC - (now - last) < 0)? 0: _t1 * ONE_SEC - (now - last);
```

Comment:
would it be clearer to say `wait = MAX(0, _t1 * ONE_SEC - (now - last))` and a definition of MAX?